pixart.js

```diff
@@ -1 +1,74 @@
+/* global $ */
 
+var inputBox = $('#color-field')
+var setColorButton = $('#set-color')
+var brushBox = $('.brush')
+
+/* button listener */
+setColorButton.on('click', function (e) {
+  e.preventDefault()
+  brushBox.css('background', inputBox.val())
+  addToSwatch(inputBox.val())
+})
+
+/* Key Enter listener */
+inputBox.on('onkeypress', function (e) {
+  if (e.which === 13) {
+    brushBox.css('background', inputBox.val())
+    addToSwatch(inputBox.val())
+  }
+})
+
+/* Build squares */
+for (var i = 0; i < 8000; i++) {
+  var $newDiv = $('<div class="square"></div>')
+  $newDiv.attr('id', 'square' + i)
+  $('body').append($newDiv)
+}
+
+/* Square listener */
+$('.square').on('mouseover', function () {
+  $(this).css('background', inputBox.val())
+})
+
+/* Bonus Material: Color Swatches */
+
+var colorSwatch = []
+
+function addToSwatch (color) {
+  // only add to swatch if it isn't already
+  // in history
+  if (colorSwatch.indexOf(color) < 0) {
+    // the length can only be up to 3
+    if (colorSwatch.length < 3) {
```

Comment:
This `if` statement can be made a bit more DRY. Notice how in both the `true` and `false` blocks we're pushing the new color and calling `rebuildSwatch()`? Since both these statements need to be called regardless of with path we take in the `if` statement, they can be pulled out and just used once outside of the `if`. Then we just tweak the condition to only perform the unique statement(s):

```js
if (colorSwatch.length >= 3) {
  colorSwatch.shift()
}
colorSwatch.push(color)
rebuildSwatch()
```